crt_portal/cts_forms/page_through.py

```diff
@@ -0,0 +1,63 @@
+from django.core.paginator import EmptyPage, PageNotAnInteger
+
+
+def pagination(paginator, page, per_page):
+    page = int(page)
+    try:
+        records = paginator.page(page)
+    except PageNotAnInteger:
+        records = paginator.page(1)
+    except EmptyPage:
+        records = paginator.page(paginator.num_pages)
+
+    adjacent_pages = 1
+
+    show_first = False
+    first_ellipsis = False
+    if page > adjacent_pages + 1:
+        show_first = True
+        first_ellipsis = True
+        if page <= adjacent_pages + 2:
+            first_ellipsis = False
```

Comment:
Could we simplify this logic down to two lines? Something like:

```
show_first = page > adjacent_pages + 1
first_ellipsis = show_first and not (page <= adjacent_pages + 2)
```

Reply:
Thanks, great suggestion!!!